### PR TITLE
fix(DTFS2-6789): allow read-only users to have access to all banks

### PR DESCRIPTION
**Pull request by lmarrai-sw**
## Introduction

When we added the `read-only` role, we thought that only the `admin` role was able to have access to all banks (when the user object has `bank.id = '*'`). However, since deploying, we have realised that the `read-only` role should be able to have access to all banks too.

## Resolution

The `validateBank` middleware in `portal` and `gef-ui` now allows users to access all banks if they have `bank.id === '*'` and the user has the `admin` _or_ `read-only` role 

## Misc

There is a `isSuperUser` function in `portal`, `portal-api`, and `gef-ui` that checks `user.bank.id === '*'` but does not check the user roles so they do not need to change for this fix

### CODE REVIEW

**Review by AlexBramhill — DISMISSED** [review]

File: gef-ui/server/middleware/validateBank/index.js [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
+const { ADMIN, READ_ONLY } = require('../../constants/roles');
 const api = require('../../services/api');
 
-const ukefAdminRoles = [ADMIN];
+const validRolesForAccessingAllBanks = [
+  ADMIN,
+  READ_ONLY,
+];
+
+const userCanAccessAllBanks = (user) => {
```
Comment (abhi-markan) [inline gef-ui/server/middleware/validateBank/index.js:9]: Can you please add function documentation.

File: gef-ui/server/middleware/validateBank/index.js [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
+  const userBankId = user?.bank?.id;
+  if (userBankId !== '*') {
```
Comment (abhi-markan) [inline gef-ui/server/middleware/validateBank/index.js:11]: Can we pick `*` from Constant?

File: portal/server/routes/middleware/validateBank/index.js [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
+  READ_ONLY,
+];
 
-const validateBank = async (req, res, next) => {
-  const { _id: dealId } = req.params;
-  const { id: bankId } = req.session.user.bank;
-  const token = req.session.userToken;
+const userCanAccessAllBanks = (user) => {
+  const userBankId = user?.bank?.id;
+  if (userBankId !== '*') {
```
Comment (abhi-markan) [inline portal/server/routes/middleware/validateBank/index.js:11]: Constant please?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
3 minor comments to address please.

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**